src/vellum/workflows/nodes/displayable/bases/inline_prompt_node/node.py

```diff
@@ -53,13 +54,18 @@ class BaseInlinePromptNode(BasePromptNode[StateType], Generic[StateType]):
     parameters: PromptParameters = DEFAULT_PROMPT_PARAMETERS
     expand_meta: Optional[AdHocExpandMeta] = OMIT
 
+    settings: Optional[PromptSettings] = None
+
     class Trigger(BasePromptNode.Trigger):
         merge_behavior = MergeBehavior.AWAIT_ANY
 
     def _get_prompt_event_stream(self) -> Iterator[AdHocExecutePromptEvent]:
         input_variables, input_values = self._compile_prompt_inputs()
         parent_context = get_parent_context()
         request_options = self.request_options or RequestOptions()
+        if self.settings and self.settings.timeout is not None:
+            request_options["timeout_in_seconds"] = int(self.settings.timeout)
```

Comment:
[blocking] ah `request_options` is already an attribute on prompt node. we could pass `self.settings` straight into the `adhoc_execute_prompt_stream ` api request below